README.md

# play_kit 

![](https://github.com/playatanu/assets/blob/main/play_kit.png?raw=true)

PLAY_KIT package containing widgets to implement glass morphism in flutter apps. You can controll Glass effect blur radius , height , blur color etc.
This package contains a set of Material Widgets that make it easy to Glass UI design. It's multi-platform, and supports mobile, desktop,
and the browser.

[![pub package](https://github.com/playatanu/assets/raw/main/buymeacoffee.png?raw=true)](https://www.buymeacoffee.com/playatanu)

## Features

- `PlayContainer`  provide a Glass Container 
- `PlayTitle`  provide a Glass Text
- `PlayDivider`  provide a Glass Divider



## Getting started


### 1. Depend on it

Add this to your package's `pubspec.yaml` file:

```yaml
dependencies:
  play_kit: ^0.0.1
```

### 2. Install it

You can install packages from the command line:

with `pub`: 

```
$ pub get play_kit
```

with `Flutter`:

```
$ flutter pub get play_kit
```

### 3. Import it

Now in your `Dart` code, you can use:

```dart
import 'package:play_kit/play_kit.dart';
```


## Using

`PlayContainer` provide a Glass `Container`:

```dart
  PlayContainer(
                height: 200,
                width: 200,
                dark: true,
                  child: PlayTitle(
                    title: 'PLAY_KIT',
                    fontSize: 30,
                    dark: true,
                    fontWeight: FontWeight.w900),
                ),
                       
```
`PlayTitle` provide a Glass `Text`:

```dart
  PlayTitle(title: 'PLAY_KIT', dark: true),
```
`PlayDivider` provide a Glass `Divider`:
```dart
  PlayDivider(height: 20, dark: true),
```


`PlayMaterial` StatelessWidget Glass UI. Include it in your `build` method like:


```dart
class PlayMaterial extends StatelessWidget {
  const PlayMaterial({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      home: Scaffold(
        body: Stack(children: [
          Image.network(
            'https://flutter.dev/assets/images/homepage/carousel/slide_1-bg.jpg',
            fit: BoxFit.cover,
            height: 800,
          ),
          Image.network(
            'https://flutter.dev/assets/images/homepage/carousel/phone_bezel.png',
            fit: BoxFit.cover,
            height: 700,
          ),
          Column(
            mainAxisAlignment: MainAxisAlignment.spaceEvenly,
            children: const [
              PlayContainer(
                height: 200,
                width: 200,
                dark: true,
                borderRadius: BorderRadius.all(Radius.circular(10)),
                child: Center(
                  child: PlayTitle(
                    title: 'PLAY_KIT',
                    fontSize: 30,
                    dark: true,
                    fontWeight: FontWeight.w900,
                  ),
                ),
              ),
              PlayDivider(
                height: 20,
                dark: true,
              ),
              PlayTitle(
                title: 'PLAY_KIT',
                fontSize: 50,
                dark: true,
                fontWeight: FontWeight.w900,
              ),
            ],
          )
        ]),
      ),
    );
  }
}

```





**Note:** You might come up with an issue that the `play_kit` does not get updated. The solution to this, to open an [issue](https://github.com/playatanu/play_kit/issues/new?template=bug_report.md).


# Bugs or Requests

If you encounter any problems feel free to open an [issue](https://github.com/playatanu/play_kit/issues/new?template=bug_report.md). If you feel the library is missing a feature, please raise a [ticket](https://github.com/playatanu/play_kit/issues/new?template=feature_request.md) on GitHub and I'll look into it. Pull request are also welcome.

This project follows the [all-contributors](https://github.com/all-contributors/all-contributors) specification. Contributions of any kind are welcome!






CHANGELOG.md
## 0.0.2

* a better example can be found in the example/ directory

## 0.0.1

* first beta version


test/play_kit_test.dart
// import 'package:flutter_test/flutter_test.dart';

// import 'package:play/play.dart';

// void main() {
//   test('adds one to input values', () {
   

//     // expect(calculator.addOne(2), 3);
//     // expect(calculator.addOne(-7), -6);
//     // expect(calculator.addOne(0), 1);
//   });
// }


example/test/widget_test.dart
// This is a basic Flutter widget test.
//
// To perform an interaction with a widget in your test, use the WidgetTester
// utility that Flutter provides. For example, you can send tap and scroll
// gestures. You can also use WidgetTester to find child widgets in the widget
// tree, read text, and verify that the values of widget properties are correct.

import 'package:flutter/material.dart';
import 'package:flutter_test/flutter_test.dart';

import 'package:play_kit_example/main.dart';

void main() {
  testWidgets('Counter increments smoke test', (WidgetTester tester) async {
    // Build our app and trigger a frame.
    await tester.pumpWidget(const PlayMaterial());

    // Verify that our counter starts at 0.
    expect(find.text('0'), findsOneWidget);
    expect(find.text('1'), findsNothing);

    // Tap the '+' icon and trigger a frame.
    await tester.tap(find.byIcon(Icons.add));
    await tester.pump();

    // Verify that our counter has incremented.
    expect(find.text('0'), findsNothing);
    expect(find.text('1'), findsOneWidget);
  });
}


example/example.dart
import 'package:flutter/material.dart';
import 'package:play_kit/play_kit.dart';

class PlayMaterial extends StatelessWidget {
  const PlayMaterial({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      home: Scaffold(
        body: Stack(children: [
          Image.network(
            'https://flutter.dev/assets/images/homepage/carousel/slide_1-bg.jpg',
            fit: BoxFit.cover,
            height: 800,
          ),
          Image.network(
            'https://flutter.dev/assets/images/homepage/carousel/phone_bezel.png',
            fit: BoxFit.cover,
            height: 700,
          ),
          Column(
            mainAxisAlignment: MainAxisAlignment.spaceEvenly,
            children: const [
              PlayContainer(
                height: 200,
                width: 200,
                dark: true,
                borderRadius: BorderRadius.all(Radius.circular(10)),
                child: Center(
                  child: PlayTitle(
                    title: 'PLAY_KIT',
                    fontSize: 30,
                    dark: true,
                    fontWeight: FontWeight.w900,
                  ),
                ),
              ),
              PlayDivider(
                height: 20,
                dark: true,
              ),
              PlayTitle(
                title: 'PLAY_KIT',
                fontSize: 50,
                dark: true,
                fontWeight: FontWeight.w900,
              ),
            ],
          )
        ]),
      ),
    );
  }
}


lib/src/playmaterial.dart
import 'dart:ui';
import 'package:flutter/material.dart';

/// [ PlayContainer ]

class PlayContainer extends StatelessWidget {
  final Widget? child;
  final double blur;
  final double height, width;
  final bool dark;
  final BorderRadiusGeometry? borderRadius;

  // ignore: use_key_in_widget_constructors
  const PlayContainer({
    this.child,
    this.blur = 5,
    required this.height,
    required this.width,
    this.dark = false,
    this.borderRadius,
  });

  @override
  Widget build(BuildContext context) {
    return ClipRRect(
      child: BackdropFilter(
        filter: ImageFilter.blur(sigmaX: blur, sigmaY: blur),
        child: Container(
          height: height,
          width: width,
          decoration: BoxDecoration(
              color: dark
                  ? Colors.black.withOpacity(0.3)
                  : Colors.white.withOpacity(0.3),
              borderRadius: borderRadius),
          child: child,
        ),
      ),
    );
  }
}

/// [ PlayDivider ]

class PlayDivider extends StatelessWidget {
  final double blur;
  final bool dark;
  @override
  // ignore: overridden_fields
  final Key? key;
  final double? height;
  final double thickness;
  final double? indent;
  final double? endIndent;

  // ignore: use_key_in_widget_constructors
  const PlayDivider({
    this.blur = 5,
    this.dark = false,
    this.key,
    this.height,
    this.thickness = 1,
    this.indent = 10,
    this.endIndent = 10,
  });

  @override
  Widget build(BuildContext context) {
    return ClipRRect(
      child: BackdropFilter(
        filter: ImageFilter.blur(sigmaX: blur, sigmaY: blur),
        child: Divider(
          color: dark
              ? Colors.black.withOpacity(0.3)
              : Colors.white.withOpacity(0.3),
          endIndent: endIndent,
          height: height,
          key: key,
          indent: indent,
          thickness: thickness,
        ),
      ),
    );
  }
}

/// [ PlayTitle ]

class PlayTitle extends StatelessWidget {
  final double blur;
  final bool dark;
  final String title;
  final double? fontSize;
  final FontWeight? fontWeight;

  // ignore: use_key_in_widget_constructors
  const PlayTitle({
    /// [PLAY TITLE]
    this.blur = 5,
    this.dark = false,
    required this.title,
    this.fontSize,
    this.fontWeight,
  });

  @override
  Widget build(BuildContext context) {
    return ClipRRect(
      child: BackdropFilter(
        filter: ImageFilter.blur(sigmaX: blur, sigmaY: blur),
        child: Text(
          title,
          style: TextStyle(
            fontSize: fontSize,
            fontWeight: fontWeight,
            color: dark
                ? Colors.black.withOpacity(0.3)
                : Colors.white.withOpacity(0.3),
          ),
        ),
      ),
    );
  }
}

/// [ PlayTextButton ]
// ignore: must_be_immutable
class PlayTextButton extends StatefulWidget {
  final Widget child;
  @override
  // ignore: overridden_fields
  final Key? key;
  void Function() onPressed;
  final double blur;
  final ButtonStyle? style;
  final bool autofocus;
  final Clip clipBehavior;
  final FocusNode? focusNode;

  final Function()? onLongPress;

  // ignore: use_key_in_widget_constructors
  PlayTextButton({
    required this.child,
    this.style,
    this.autofocus = false,
    this.onLongPress,
    required this.onPressed,
    this.key,
    this.blur = 5,
    this.clipBehavior = Clip.none,
    this.focusNode,
  });

  @override
  State<PlayTextButton> createState() => _PlayTextButtonState();
}

class _PlayTextButtonState extends State<PlayTextButton> {
  @override
  Widget build(BuildContext context) {
    return ClipRRect(
      child: BackdropFilter(
        filter: ImageFilter.blur(sigmaX: widget.blur, sigmaY: widget.blur),
        child: TextButton(
          style: widget.style,
          autofocus: widget.autofocus,
          onLongPress: widget.onLongPress,
          key: widget.key,
          clipBehavior: widget.clipBehavior,
          focusNode: widget.focusNode,
          onPressed: widget.onPressed,
          child: widget.child,
        ),
      ),
    );
  }
}


example/windows/runner/main.cpp
#include <flutter/dart_project.h>
#include <flutter/flutter_view_controller.h>
#include <windows.h>

#include "flutter_window.h"
#include "utils.h"

int APIENTRY wWinMain(_In_ HINSTANCE instance, _In_opt_ HINSTANCE prev,
                      _In_ wchar_t *command_line, _In_ int show_command) {
  // Attach to console when present (e.g., 'flutter run') or create a
  // new console when running with a debugger.
  if (!::AttachConsole(ATTACH_PARENT_PROCESS) && ::IsDebuggerPresent()) {
    CreateAndAttachConsole();
  }

  // Initialize COM, so that it is available for use in the library and/or
  // plugins.
  ::CoInitializeEx(nullptr, COINIT_APARTMENTTHREADED);

  flutter::DartProject project(L"data");

  std::vector<std::string> command_line_arguments =
      GetCommandLineArguments();

  project.set_dart_entrypoint_arguments(std::move(command_line_arguments));

  FlutterWindow window(project);
  Win32Window::Point origin(10, 10);
  Win32Window::Size size(1280, 720);
  if (!window.CreateAndShow(L"play_kit_example", origin, size)) {
    return EXIT_FAILURE;
  }
  window.SetQuitOnClose(true);

  ::MSG msg;
  while (::GetMessage(&msg, nullptr, 0, 0)) {
    ::TranslateMessage(&msg);
    ::DispatchMessage(&msg);
  }

  ::CoUninitialize();
  return EXIT_SUCCESS;
}


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:play_kit/play_kit.dart';

void main() {
  runApp(const PlayMaterial());
}

class PlayMaterial extends StatelessWidget {
  const PlayMaterial({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      home: Scaffold(
        body: Stack(children: [
          Image.network(
            'https://flutter.dev/assets/images/homepage/carousel/slide_1-bg.jpg',
            fit: BoxFit.cover,
            height: 800,
          ),
          Image.network(
            'https://flutter.dev/assets/images/homepage/carousel/phone_bezel.png',
            fit: BoxFit.cover,
            height: 700,
          ),
          Column(
            mainAxisAlignment: MainAxisAlignment.spaceEvenly,
            children: const [
              PlayContainer(
                height: 200,
                width: 200,
                dark: true,
                borderRadius: BorderRadius.all(Radius.circular(10)),
                child: Center(
                  child: PlayTitle(
                    title: 'PLAY_KIT',
                    fontSize: 30,
                    dark: true,
                    fontWeight: FontWeight.w900,
                  ),
                ),
              ),
              PlayDivider(
                height: 20,
                dark: true,
              ),
              PlayTitle(
                title: 'PLAY_KIT',
                fontSize: 50,
                dark: true,
                fontWeight: FontWeight.w900,
              ),
            ],
          )
        ]),
      ),
    );
  }
}
